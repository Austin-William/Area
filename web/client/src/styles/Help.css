/* This file change the display of Help toast */

.Help {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    color: whitesmoke;
}

.Help-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 2em;
    color: rgb(0, 0, 0);
}

/* Help's content CSS part */

.Help-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.Help-content .Help-content-item-title {
    width: 100%;
    height: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 0.5em;
    background-color: rgb(0, 0, 0);
}

.Help-content .Help-content-item {
    overflow-y: scroll;
    width: 100%;
    height: 50em;
    margin: 0.5em;
    padding: 0.5em;
    border-radius: 0.5em;
    background-color: rgb(255, 255, 255);
}

.Help-content .Help-content-item-title h3 {
    margin: 0;
    padding: 0;
    font-size: 1.5em;
    color: whitesmoke;
}

/* CSS part for responsive design */

@media screen and (max-width: 768px) {
    .Help-container {
        width: 100%;
        height: 100%;
        padding: 0.5em;
    }

    .Help-content .Help-content-item {
        width: 100%;
    }

    .Help-content .Help-content-item h3 {
        font-size: 1.2em;
        color: grey;
    }
}

/* CSS for scroll bar */

::-webkit-scrollbar {
    width: 10px;
}

::-webkit-scrollbar-track {
    background: linear-gradient(30deg, #67a8d3 40%, rgb(99, 134, 209) 60%);
    border-radius: 10px;
}

::-webkit-scrollbar-thumb {
    background: linear-gradient(30deg, #61ebf5 20%, #85f5fd 100%);
    border-radius: 3px;
}

@media (max-width: 768px) {
    .Home-container .Home-section {
        grid-template-columns: 1fr;
        margin: 1em;
    }
}