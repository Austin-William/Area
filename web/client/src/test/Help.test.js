import React from "react";
import renderer from "react-test-renderer";
import Help from "../pages/Help";

test("Pages : Help renders correctly", () => {
  const tree = renderer.create(<Help />).toJSON();
  expect(tree).toMatchSnapshot();
});

