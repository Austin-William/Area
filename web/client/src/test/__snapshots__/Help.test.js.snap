// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages : Help renders correctly 1`] = `
<div
  className="Help"
>
  <div
    className="Help-container"
  >
    <div
      className="Help-content"
    >
      <div
        className="Help-content-item"
      >
        <h2>
          How to use the widgets?
        </h2>
        <br />
        <ul>
          <li>
            To connect to the widget, click on the "Add API" button and connect your accounts.
          </li>
          <li>
            To receive emails, go to the settings page and add your email in the input field. Dont forget to add the ip adress of your server. (ex: http://localhost:8080)
          </li>
        </ul>
        <br />
        <br />
        <h2>
          How to process the action/reaction
        </h2>
        <br />
        <ul>
          <li>
            First, choose the reaction type before chossing action type.
          </li>
          <li>
            Then, choose the action type. Clicking on the submit button will send the action to the server.
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
